jetbot/bot.py

```diff
@@ -69,7 +69,42 @@ def product_url(self, data: str) -> str:
     def issue_url(self, data: str) -> str:
         return "<https://youtrack.jetbrains.com/issues/" + data + ">"
 
-    # Find a product channel
+    # Find product channels
+    async def product_channels(self, data: dict) -> list[TextChannel]:
+        channels = []
+        if not data.get("category_name"):
+            return channels
+
+        guild = self.get_guild(self.config.guild)
+        if not guild:
+            guild = await self.fetch_guild(self.config.guild)
+
+        text_channels = guild.text_channels
+        if not text_channels:
+            text_channels = await guild.fetch_channels()
+            text_channels = [f for f in text_channels if isinstance(f, TextChannel)]
+
+        # Handle multi-channel configuration
+        if "channels" in data:
+            for channel_config in data["channels"]:
+                for channel in text_channels:
+                    if channel.name == channel_config["name"] \
+                            and channel.category \
+                            and channel.category.name.lower().strip() == data["category_name"].lower().strip():
+                        channels.append(channel)
+                        break
+        # Handle legacy single-channel configuration
+        elif data.get("channel_name"):
+            for channel in text_channels:
+                if channel.name == data["channel_name"] \
+                        and channel.category \
+                        and channel.category.name.lower().strip() == data["category_name"].lower().strip():
+                    channels.append(channel)
+                    break
```

Comment:
Instead of handling both cases with near-identical code, can we force ourselves to always have a list and then loop over it?

```suggestion
        config_channels = data.get("channels", [data["channel"])])
        for channel_config in config_channels:
            for channel in text_channels:
                if channel.name == channel_config["name"] \
                        and channel.category \
                        and channel.category.name.lower().strip() == data["category_name"].lower().strip():
                    channels.append(channel)
                    break
```